dashboard/src/components/Catalog/Catalog.tsx

```diff
@@ -77,16 +97,96 @@ class Catalog extends React.Component<ICatalogProps, ICatalogState> {
           />
         </PageHeader>
         <LoadingWrapper loaded={!isFetching}>
-          <CardGrid>{chartItems}</CardGrid>
+          <div className="row">
+            {csvs.length > 0 && (
+              <div className="col-2">
+                <div className="margin-b-normal">
+                  <span>
+                    <b>Type:</b>
+                  </span>
+                </div>
+                <div>
+                  <label className="checkbox" key="listcharts">
+                    <input type="checkbox" checked={listCharts} onChange={this.toggleListCharts} />
+                    <span>Charts</span>
+                  </label>
+                </div>
+                <div>
+                  <label className="checkbox" key="listoperators">
+                    <input
+                      type="checkbox"
+                      checked={listOperators}
+                      onChange={this.toggleListOperators}
+                    />
+                    <span>Operator Instances</span>
+                  </label>
+                </div>
+              </div>
+            )}
+            <div className={csvs.length > 0 ? "col-10" : ""}>
+              <CardGrid>{items}</CardGrid>
+            </div>
+          </div>
         </LoadingWrapper>
       </section>
     );
   }
 
-  private filteredCharts(charts: IChart[], filter: string) {
+  private filteredCharts = (charts: IChart[]) => {
+    const { filter, listCharts } = this.state;
+    if (!listCharts) {
+      return [];
+    }
     return charts.filter(c => new RegExp(escapeRegExp(filter), "i").test(c.id));
+  };
+
+  private filteredCSVs(csvs: IClusterServiceVersion[]) {
+    const { filter, listOperators } = this.state;
+    if (!listOperators) {
+      return [];
+    }
+    return csvs.filter(c => new RegExp(escapeRegExp(filter), "i").test(c.metadata.name));
   }
 
+  private getCatalogItems(charts: IChart[], csvs: IClusterServiceVersion[]): ICatalogItem[] {
+    let result: ICatalogItem[] = [];
+    charts.forEach(c => {
+      result = result.concat({
+        id: c.id,
+        name: c.attributes.name,
+        icon: c.attributes.icon ? `api/assetsvc/${c.attributes.icon}` : undefined,
+        version: c.relationships.latestChartVersion.data.app_version,
+        description: c.attributes.description,
+        type: "chart",
+        repoName: c.attributes.repo.name,
+        namespace: this.props.namespace,
+      });
+    });
+    csvs.forEach(csv => {
+      csv.spec.customresourcedefinitions.owned.forEach(crd => {
+        result = result.concat({
+          id: crd.name,
+          name: crd.displayName,
+          icon: `data:${csv.spec.icon[0].mediatype};base64,${csv.spec.icon[0].base64data}`,
+          version: crd.version,
+          description: crd.description,
+          type: "operator",
+          csv: csv.metadata.name,
+          namespace: this.props.namespace,
+        });
+      });
+    });
+    return result;
```

Comment:
Looks like this is returning all charts followed by all operators? Should the catalog result be sorted in some way?

Reply:
It is. Probably is more intuitive to have them shuffled and ordered alphabetically